substrate/primitives/staking/src/lib.rs

```diff
@@ -368,37 +370,43 @@ pub struct Exposure<AccountId, Balance: HasCompact> {
 	pub others: Vec<IndividualExposure<AccountId, Balance>>,
 }
 
-impl<AccountId, Balance: Default + HasCompact> Default for Exposure<AccountId, Balance> {
+impl<AccountId, Balance: Default + HasCompact + MaxEncodedLen> Default
+	for Exposure<AccountId, Balance>
+{
 	fn default() -> Self {
 		Self { total: Default::default(), own: Default::default(), others: vec![] }
 	}
 }
 
-impl<
-		AccountId: Clone,
-		Balance: HasCompact + AtLeast32BitUnsigned + Copy + codec::MaxEncodedLen,
-	> Exposure<AccountId, Balance>
+impl<AccountId: Clone, Balance: HasCompact + AtLeast32BitUnsigned + Copy + MaxEncodedLen>
+	Exposure<AccountId, Balance>
 {
 	/// Splits an `Exposure` into `PagedExposureMetadata` and multiple chunks of
 	/// `IndividualExposure` with each chunk having maximum of `page_size` elements.
-	pub fn into_pages(
+	pub fn into_pages<MaxExposurePageSize>(
```

Comment:
This function is only used one time and `page_size` is equal to `MaxExposurePageSize`. I think we should refactor: remove the argument page_size.

If needed we can always introduce a try_into_pages with an argument.
In general I feel it is not so good to add many implicit constraint, like here `page_size` must be less than `MaxExposurePageSize` otherwise some element will be ignored (with a log::error, but still).

Reply:
Changed in https://github.com/paritytech/polkadot-sdk/pull/6445/commits/489fc9ddc092f1fad2876551b4a3254096bc29d8